apps/meteor/ee/server/apps/communication/rest.ts

```diff
@@ -1124,14 +1128,48 @@ export class AppsRestApi {
 						return API.v1.failure('Invalid status provided, it must be "status" field and a string.');
 					}
 
-					const prl = manager.getOneById(this.urlParams.id);
+					const baseUrl = orchestrator.getMarketplaceUrl();
+					const { id: appId } = this.urlParams;
+					const { status } = this.bodyParams;
 
+					const headers = getDefaultHeaders();
+
+					const prl = manager.getOneById(appId);
 					if (!prl) {
-						return API.v1.notFound(`No App found by the id of: ${this.urlParams.id}`);
+						return API.v1.notFound(`No App found by the id of: ${appId}`);
+					}
+
+					const storedApp = prl.getStorageItem();
+
+					const { version } = prl.getInfo();
+
+					const { isEnterpriseOnly: isStoredAppEnterpriseOnly } = storedApp.marketplaceInfo as IMarketplaceInfo;
+
+					let isAppEnterpriseOnly = isStoredAppEnterpriseOnly ?? false;
+
+					if (!isEnterprise()) {
+						const appInfosURL = new URL(`${baseUrl}/v1/apps/${appId}`);
+
+						appInfosURL.searchParams.set('appVersion', String(version));
+						const appInfoResponse = await fetch(appInfosURL.toString(), {
+							headers,
+						});
+
+						if (appInfoResponse?.ok) {
+							const [data] = await appInfoResponse.json();
+
+							isAppEnterpriseOnly = data?.isEnterpriseOnly;
+						}
+
+						if (![AppStatus.DISABLED, AppStatus.MANUALLY_DISABLED].includes(status)) {
+							if (isAppEnterpriseOnly) {
```

Comment:
There doesn't seem to be any benefits on splitting this if statement. Please prefer using the more concise `&` operator